### PR TITLE
Create a playground to locally preview SXG

### PR DESCRIPTION
**Pull request by antiphoton**

### CODE REVIEW

File: playground/src/server/credentials.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    const privateKeyFile = path.join(certDir, 'privkey.pem');
+    await fs.writeFile(privateKeyFile, privateKeyPem);
+    const privateKeyJwk = crypto.createPrivateKey(privateKeyPem)
+        .export({format: 'jwk'});
+    const certificateRequestFile = path.join(certDir, 'csr.pem');
+    const {
+        stdout: certificateRequestPem,
+    } = await execFile(
+        'openssl',
+        ['req', '-new', '-sha256', '-key', privateKeyFile, '-subj', `/CN=${domain}/OTest/C=US`],
```
Comment (twifkak) [inline playground/src/server/credentials.ts:45]: I think you mean `O=Test`, but actually may just omit `/OTest/C=US` as it's not necessary for the test.

File: playground/src/server/index.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    const worker = await createWorker(wasmBuffer, SXG_CONFIG, certificatePem, certificatePem);
+    const sxgList: WasmResponse[] = [];
+    async function createSxgIntoList(innerUrl: string, certOrigin: string) {
+        const sxgPayload = await(fetcher({
+            url: innerUrl,
+            body: [],
+            method: 'Get',
+            headers: [],
+        }));
+        const sxg = await worker.createSignedExchange(
```
Comment (twifkak) [inline playground/src/server/index.ts:89]: Is it possible to call `handleRequest` from cloudflare_worker/worker/src/index.ts instead? Or alternatively replicate the `processHTML` calls that it does. OK to defer this to a future PR.

File: playground/src/server/wasmFunctions.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  shouldRespondDebugInfo(): boolean;
+  validatePayloadHeaders(fields: HeaderFields): void,
+}
+
+interface WasmFunctions {
+  init: () => void,
+  WasmWorker: WasmWorker,
+}
+
+export async function createWorker(wasmBytes: Buffer, configYaml: string, certPem: string, issuerPem: string) {
```
Comment (twifkak) [inline playground/src/server/wasmFunctions.ts:79]: Can you leave a comment that this is the only part that differs from `cloudflare_worker/worker/src/wasmFunctions.ts`, and mention why (e.g. to allow parameterizable config/cert/issuer)?

(Ideally we wouldn't have the duplication, but I'm okay deferring that work as long as we note that it exists.)

**Review by twifkak — APPROVED** [review]

File: playground/src/server/credentials.ts (hunk 1 above)
Comment (antiphoton) [reply playground/src/server/credentials.ts]: Removed `/OTest/C=US`.

File: playground/src/server/index.ts (hunk 2 above)
Comment (antiphoton) [reply playground/src/server/index.ts:90]: Will do this in a future PR.

File: playground/src/server/wasmFunctions.ts (hunk 3 above)
Comment (antiphoton) [reply playground/src/server/wasmFunctions.ts:81]: Added a TODO at top of this file.

**Pull request merged**